code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -888,6 +888,32 @@
 		return
 
 /mob/living/silicon/robot/proc/self_destruct()
+	say("WARNING: Self-destruct initiated. Unit [src] will self destruct in five seconds.")
+	lockcharge = 1
+	update_canmove()
+	sleep(20)
+	playsound(loc, 'sound/items/countdown.ogg', 125, 1)
+	sleep(20)
+	playsound(loc, 'sound/effects/alert.ogg', 125, 1)
+	sleep(10)
+	var/turf/O = get_turf(src)
+	var/fragger = rand(50,100)
+	density = 0
+	explosion(O, -1, -1, 2)
+	var/list/target_turfs = getcircle(O, 7)
+	var/fragments_per_projectile = round(fragger/target_turfs.len)

```

Comment:
I think it'd be a better idea to create a reuseable effect object to fire shrapnel all around

Having snowflake code in everything that should throw shrapnel doesn't seem like a good approach